server/src/main/java/org/elasticsearch/action/admin/cluster/snapshots/globalstate/SnapshotGlobalStateResponse.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.action.admin.cluster.snapshots.globalstate;
+
+import org.elasticsearch.action.ActionResponse;
+import org.elasticsearch.cluster.metadata.Metadata;
+import org.elasticsearch.common.collect.Iterators;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.xcontent.ChunkedToXContentHelper;
+import org.elasticsearch.common.xcontent.ChunkedToXContentObject;
+import org.elasticsearch.xcontent.ToXContent;
+
+import java.io.IOException;
+import java.util.Iterator;
+
+public class SnapshotGlobalStateResponse extends ActionResponse implements ChunkedToXContentObject {
+
+    private final Metadata metadata;
+
+    public SnapshotGlobalStateResponse(StreamInput in) throws IOException {
+        super(in);
+        metadata = in.readOptionalWriteable(Metadata::readFrom);
```

Comment:
This shouldn't be optional I think, it's never `null` in practice. If the metadata is missing then we throw an exception trying to load it instead.

```suggestion
        metadata = in.readWriteable(Metadata::readFrom);
```

Reply:
After adding `TransportVersion`, I am reading it as below:

```
public SnapshotGlobalStateResponse(StreamInput in) throws IOException {
        super(in);
        if (in.getTransportVersion().onOrAfter(GLOBAL_STATE_VERSION)) {
            metadata = Metadata.readFrom(in);
        } else {
            metadata = Metadata.EMPTY_METADATA;
            throw new IllegalArgumentException();
        }

    }
```